src/foam/nanos/fs/fileDropZone/FileDropZone.js

```diff
@@ -203,47 +197,49 @@ foam.CLASS({
-          .start()
-            .start('p').addClass('p-xs', this.myClass('caption')).add(this.LABEL_SUPPORTED).end()
-            .start('p').addClass('p-xs', self.myClass('caption')).add(this.getSupportedTypes(true)).end()
-          .end()
-          .start()
-            .start('p').addClass('p-xs', this.myClass('caption')).add(this.LABEL_MAX_SIZE + ' ' + this.maxSize + 'MB').end()
+        .callIf(this.controllerMode != foam.u2.ControllerMode.VIEW, function() {
+          this.start().addClass(this.myClass())
+            .addClass(this.myClass('instruction-container'))
+            .enableClass('selection', this.hasFiles$)
+            .enableClass(this.myClass('dragged'), this.isDragged_$)
+            .start()
+              .addClass(this.myClass('browse-container'))
+              .enableClass(this.myClass('browse-container-row'), this.hasFiles$)
+              .start().addClass('p-semiBold').add(this.title || this.LABEL_DEFAULT_TITLE).end()
+              .start().addClass('p').add(this.LABEL_OR).end()
+              .add(this.slot(function(hasFiles) {
+                return this.E().start(this.BROWSE, {
+                  label: self.LABEL_BROWSE,
+                  buttonStyle: hasFiles ? 'LINK' : 'SECONDARY'
+                })
+                  .enableClass(this.myClass('link'), self.hasFiles$)
+                  .attrs({
+                    for: 'file-upload'
+                  })
+                  .end();
+              }))
+            .end()
+
+            .start().addClass(this.myClass('caption-container'))
+              .show(this.slot(function(showHelp, files) { return showHelp && files.length < 1; }))
+              .start()
+                .start('p').addClass('p-xs', this.myClass('caption')).add(this.LABEL_SUPPORTED).end()
+                .start('p').addClass('p-xs', self.myClass('caption')).add(this.getSupportedTypes(true)).end()
+              .end()
+              .start()
+                .start('p').addClass('p-xs', this.myClass('caption')).add(this.LABEL_MAX_SIZE + ' ' + this.maxSize + 'MB').end()
+              .end()
+            .end()
           .end()
-        .end()
-      .end()
-      .on('drop', this.onDrop)
-      .on('dragover', e => { this.isDragged_ = true; e.preventDefault(); } )
-      .on('dragenter', e => { this.isDragged_ = true; e.preventDefault(); })
-      .on('dragleave', e => { this.isDragged_ = false; e.preventDefault(); })
-      .add(this.slot(function(files) {
+          .on('drop', this.onDrop)
+          .on('dragover', e => { this.isDragged_ = true; e.preventDefault(); } )
+          .on('dragenter', e => { this.isDragged_ = true; e.preventDefault(); })
+          .on('dragleave', e => { this.isDragged_ = false; e.preventDefault(); });
+        })
+        .add(this.slot(function(files) {
         var e = this.E().addClass(self.myClass('fileCards'));
-        if ( ! visibilitySlot.get() )
+        if ( files.length == 0 )
```

Comment:
Shouldn't this be the only change required? Other changes make this view require a full rerender on controller mode change. Myabe replace this `.add()` with
` .add(this.slot(function(files, visibility) {
        var e = this.E().addClass(self.myClass('fileCards'));
        if ( ! files.length && ! visibility )
          return e.add(self.NO_FILES);
        for ( var i = 0; i < files.length; i++ ) {
          e.tag({
            class: 'foam.nanos.fs.fileDropZone.FileCard',
            data: files[i],
            selected: this.selected,
            index: i
          });
        }
        return e;
      }, this.files$, visibilitySlot));
      ` to also not show the no file logo in Edit view

Reply:
Yes, that worked.  Thanks